cirq/ops/common_channels.py

```diff
@@ -50,90 +60,156 @@ def __init__(self, p_x: float, p_y: float, p_z: float) -> None:
             $$
 
         Args:
+            num_qubits: Number of qubits
+            pis: Dictionary of string (Pauli operator) to its probability
             p_x: The probability that a Pauli X and no other gate occurs.
             p_y: The probability that a Pauli Y and no other gate occurs.
             p_z: The probability that a Pauli Z and no other gate occurs.
 
         Raises:
             ValueError: if the args or the sum of args are not probabilities.
         """
-
-        self._p_x = value.validate_probability(p_x, 'p_x')
-        self._p_y = value.validate_probability(p_y, 'p_y')
-        self._p_z = value.validate_probability(p_z, 'p_z')
-        self._p_i = 1 - value.validate_probability(p_x + p_y + p_z,
-                                                   'p_x + p_y + p_z')
+        assert ('pis' in kwargs or 'p_x' in kwargs or 'p_y' in kwargs or
+                'p_z' in kwargs or len(args) > 0)
+        if 'pis' in kwargs:
+            assert 'num_qubits' in kwargs
```

Comment:
Instead of asserting, we should raise `ValueError` when `ps` contains a key whose length is different than `num_qubits`. Note that it is ok not to specify `num_qubits` when we're creating a single-qubit channel.

Reply:
Done.